app/src/components/QuestMap.vue

```diff
@@ -73,9 +75,24 @@ const genOption = (nodes, edges) => ({
       itemStyle: {
         color: (params) => {
           const {
-            data: { completed },
+            data: { completed, unlocked },
           } = params
-          return completed ? '#ADADB5' : '#454552'
+          switch (completed !== unlocked) {
+            case false: {
+              // completed==unlocked, when completed or not yet unlocked
+              if (completed) {
+                return '#00ff00'
+              }
+              return '#454552'
+            }
+            case true: {
+              // completed!=unlocked, only when unlocked but not completed
+              return '#ADADB5'
+            }
+            default: {
+              return '#454552'
+            }
+          }
```

Comment:
Don't use switch to check boolean value, use if-else or ternary operator instead (`a ? b : c`)
```suggestion
          if (completed) {
            return '#00ff00'
          }
          if (unlocked) {
            return '#ADADB5' 
          }
          return '#454552'
```